core/common/assets/js/api/core/store.js

```diff
@@ -0,0 +1,125 @@
+import { configureStore, combineReducers } from '@reduxjs/toolkit';
+
+export default class Store {
+	/**
+	 * Initialize the Store.
+	 *
+	 * @return {void}
+	 */
+	constructor() {
+		this.slices = {};
+		this.reduxStore = this.createStore();
+	}
+
+	/**
+	 * Create a Redux Store object.
+	 *
+	 * @return {EnhancedStore<{}, AnyAction, [ThunkMiddlewareFor<{}>]>}
+	 */
+	createStore() {
+		const store = configureStore( {
+			// Use an empty function instead of empty object since an empty object
+			// isn't a valid reducer value.
+			reducer: () => {},
+		} );
+
+		// See: https://redux.js.org/usage/code-splitting#defining-an-injectreducer-function
+		store.asyncReducers = {};
+
+		store.injectReducer = ( key, newReducer ) => {
+			store.asyncReducers[ key ] = newReducer;
+
+			store.replaceReducer( combineReducers( { ...store.asyncReducers } ) );
+		};
```

Comment:
```suggestion
		return configureStore( {
			// Use an empty function instead of empty object since an empty object
			// isn't a valid reducer value.
			reducer: () => {},
		} );
```

Reply:
Done